### PR TITLE
[wanda] main forge binary

### PR DESCRIPTION
**Pull request by aslonnie**
this makes wanda binary able to build a wanda yaml file, and can work with work repos.

### CODE REVIEW

File: wanda/forge.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+			remote.WithAuthFromKeychain(authn.DefaultKeychain),
+		},
+	}, nil
+}
+
+func (f *Forge) addSrcFile(ts *tarStream, src string) {
+	ts.addFile(src, nil, filepath.Join(f.workDir, src))
+}
+
+func (f *Forge) workTag(name string) string {
```
Comment (can-anyscale) [inline wanda/forge.go:71]: what is the tag used for?

File: wanda/forge.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+	// TODO(aslonnie): check if the image output already exists
+	// if yes, then just perform retag, rather than rebuilding.
+
+	// Get all the tags.
+
+	// Work tag is the tag we use to save the image in the work repo.
+	var workTag string
+	if f.config.WorkRepo != "" {
+		if f.config.BuildID != "" {
```
Comment (can-anyscale) [inline wanda/forge.go:157]: there was a function for this above

**Review by can-anyscale — APPROVED** [review]

File: wanda/forge.go (hunk 2 above)
Comment (aslonnie) [reply wanda/forge.go:157]: yes, nice catch.

File: wanda/forge.go (hunk 1 above)
Comment (aslonnie) [reply wanda/forge.go:71]: this is the tag / image name to be saved in the work repo, prefixed with build id.

**Pull request merged**